velox/exec/RowContainer.cpp

```diff
@@ -336,16 +334,38 @@ char* RowContainer::initializeRow(char* row, bool reuse) {
   return row;
 }
 
+/**
+ * Removes the data of a given row from the column stats by iterating
+ * over each column in the row and updates the column stats.
+ *
+ * @param row - The row from which the column stats are to be updated.
+ */
+void RowContainer::removeRowFromColumnStats(const char* row) {
+  // Update row column stats accordingly
+  for (auto c = 0; c < keyTypes_.size(); ++c) {
+    auto byteSize = typeKindSize(typeKinds_[c]);
+    if (isNullAt(row, columnAt(c))) {
+      rowColumnsStats_[c].removeCellSize(0, true);
+    } else if (types_[c]->isFixedWidth()) {
+      rowColumnsStats_[c].removeCellSize(fixedSizeAt(c), false);
+    } else {
+      rowColumnsStats_[c].removeCellSize(variableSizeAt(row, c), false);
+    }
+  }
+  invalidateMinMaxColumnStats();
+}
+
 void RowContainer::eraseRows(folly::Range<char**> rows) {
   freeRowsExtraMemory(rows, /*freeNextRowVector=*/true);
   for (auto* row : rows) {
     VELOX_CHECK(!bits::isBitSet(row, freeFlagOffset_), "Double free of row");
     bits::setBit(row, freeFlagOffset_);
     nextFree(row) = firstFreeRow_;
     firstFreeRow_ = row;
+
+    removeRowFromColumnStats(row);
```

Comment:
```
s/removeRowFromColumnStats/removeRowColumnStats/
```

Reply:
Hmm... it is actually meant to remove some rows info from the rowColumnStats. removeRowColumnStats sounds like we removed all the stats. wdyt?